pkg/logs/processor/processor.go

```diff
@@ -172,22 +171,20 @@ func (p *Processor) applySDSReconfiguration(order sds.ReconfigureOrder) {
 	if err != nil {
 		log.Errorf("Error while reconfiguring the SDS scanner: %v", err)
 	} else {
-		// no error while reconfiguring the SDS scanner and it is active?
+		// no error while reconfiguring the SDS scanner and since it looks active now,
 		// we should drain the buffered messages if any and stop the
 		// buffering mechanism.
 		if p.sds.buffering && isActive {
-			if p.sds.waitForConfig {
-				log.Debug("Processor ready with an SDS configuration.")
-				p.sds.buffering = false
+			log.Debug("Processor ready with an SDS configuration.")
+			p.sds.buffering = false
 
-				// drain the buffer of messages if anything's in there
-				log.Info("SDS: draining", len(p.sds.buffer), "buffered messages")
-				for _, msg := range p.sds.buffer {
-					p.processMessage(msg)
-				}
-
-				p.sds.resetBuffer()
+			// drain the buffer of messages if anything's in there
+			log.Info("SDS: sending", len(p.sds.buffer), "buffered messages")
+			for _, msg := range p.sds.buffer {
+				p.processMessage(msg)
```

Comment:
I thought I asked already, but can't find it. Sorry if duplicate.

Is it possible that this takes so long that the RC request times out and misleadingly report an error?

Reply:
> Is it possible that this takes so long

What are you thinking about? (i.e. what could take so long, draining the buffer?)

Reply:
Yes, AIUI the outgoing pipeline capacity is limited and fairly small, draining any non-trivial amount of logs would get blocked on sending data through the network, which can take a long time. Sending 500MB may take a while even in best conditions. Can it cause any issues?

Reply:
It can possibly induce drops in the listener. While this PR is a workaround for starting agent, I don't expect to buffer more than a few MBs here to be completely honest, this buffering mechanism isn't a full solution for a long unreachable RC delivery. I'll document this in the pull-request description. OTOT, we will recommend the `no_collection` behaviour every time it's possible, which is most likely to be all the time (e.g. no TCP listener).

Reply:
> It can possibly induce drops in the listener

Remote Config listener? Could you elaborate a bit please?

> I don't expect to buffer more than a few MBs here to be completely honest

Would it make sense to set the buffer size accordingly?

> this buffering mechanism isn't a full solution for a long unreachable RC delivery

My concern here is opposite: backpressure from logs (e.g. from slow network or misconfigured proxy) should not negatively affect RC.

Reply:
Sorry I wasn't clear enough: the TCP listener. Slow processing of megs of data would apply backpressure on the logs tailers, RC isn't part of this picture and would not be impacted (the client is in its goroutine and poll-based).

Reply:
Discussed offline: the current buffer implementation could indeed block the RC client for the time of processing the buffered messages (i.e. seconds)

Added this to the PR description:
> If the buffer mechanism is used to unpile and process a lot of buffered messages, it could block the remote configuration client for a few seconds.